### PR TITLE
test(spanner): fix flaky restore_database() sample test

### LINKED ISSUE #4544: spanner.cloud-client.backup_sample_test: test_restore_database failed
Note: #3238 was also for this test, but it was closed more than 10 days ago. So, I didn't mark it flaky.

----

commit: 335886ab2da68426b00a56fa7abd33ce1c733b4b
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/aef419da-842c-48c2-9641-1d84a2b02dbe), [Sponge](http://sponge2/aef419da-842c-48c2-9641-1d84a2b02dbe)
status: failed
<details><summary>Test output</summary><br><pre>Traceback (most recent call last):
  File "/workspace/spanner/cloud-client/.nox/py-3-7/lib/python3.7/site-packages/google/api_core/retry.py", line 184, in retry_target
    return target()
  File "/workspace/spanner/cloud-client/.nox/py-3-7/lib/python3.7/site-packages/google/api_core/future/polling.py", line 84, in _done_or_raise
    raise _OperationNotComplete()
google.api_core.future.polling._OperationNotComplete

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/workspace/spanner/cloud-client/.nox/py-3-7/lib/python3.7/site-packages/google/api_core/future/polling.py", line 104, in _blocking_poll
    retry_(self._done_or_raise)()
  File "/workspace/spanner/cloud-client/.nox/py-3-7/lib/python3.7/site-packages/google/api_core/retry.py", line 286, in retry_wrapped_func
    on_error=on_error,
  File "/workspace/spanner/cloud-client/.nox/py-3-7/lib/python3.7/site-packages/google/api_core/retry.py", line 206, in retry_target
    last_exc,
  File "<string>", line 3, in raise_from
google.api_core.exceptions.RetryError: Deadline of 1200.0s exceeded while calling functools.partial(<bound method PollingFuture._done_or_raise of <google.api_core.operation.Operation object at 0x7f8f08c19750>>), last exception: 

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/workspace/spanner/cloud-client/backup_sample_test.py", line 71, in test_restore_database
    backup_sample.restore_database(INSTANCE_ID, RESTORE_DB_ID, BACKUP_ID)
  File "/workspace/spanner/cloud-client/backup_sample.py", line 71, in restore_database
    operation.result(1200)
  File "/workspace/spanner/cloud-client/.nox/py-3-7/lib/python3.7/site-packages/google/api_core/future/polling.py", line 125, in result
    self._blocking_poll(timeout=timeout)
  File "/workspace/spanner/cloud-client/.nox/py-3-7/lib/python3.7/site-packages/google/api_core/future/polling.py", line 107, in _blocking_poll
    "Operation did not complete within the designated " "timeout."
concurrent.futures._base.TimeoutError: Operation did not complete within the designated timeout.</pre></details>

### PR DESCRIPTION
**Pull request by larkee**
Fixes #4544

### CODE REVIEW

**Review by skuruppu — APPROVED** [review]

**Review by skuruppu — APPROVED** [review]

File: spanner/cloud-client/backup_sample_test.py [hunk 1]
```diff
@@ -13,6 +13,8 @@
 # limitations under the License.
 import uuid
 
+from google.api_core.exceptions import DeadlineExceeded
+from test_utils.retry import RetryErrors
```
Comment (tmatsuo) [inline spanner/cloud-client/backup_sample_test.py:17]: Where is `test_utils`?

File: spanner/cloud-client/backup_sample_test.py [hunk 2]
```diff
@@ -67,6 +67,13 @@ def test_create_backup(capsys, database):
     assert BACKUP_ID in out
 
 
+@pytest.mark.skip(
```
[the remark this answers is not in the document]
Comment (kurtisvg) [reply spanner/cloud-client/backup_sample_test.py]: > @kurtisvg Is here a way to extend the timeout for a particular test?

Nope

File: spanner/cloud-client/backup_sample_test.py (hunk 1 above)
Comment (larkee) [reply spanner/cloud-client/backup_sample_test.py]: Whoops. It's a library used in the Cloud Spanner tests. I've added it as an import.

File: spanner/cloud-client/backup_sample_test.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (larkee) [reply spanner/cloud-client/backup_sample_test.py]: I've used a RetryErrors decorator from the Cloud Spanner test_utils library which will retry the test on `DEADLINE_EXCEEDED` for a maximum of two attempts. Is this an acceptable alternative to `flaky`?

**Review by tmatsuo — APPROVED** [review]

**Pull request merged**